python/pyspark/sql/tests/test_arrow.py

```diff
@@ -421,6 +421,35 @@ def run_test(num_records, num_parts, max_records, use_delay=False):
             run_test(*case)
 
 
+@unittest.skipIf(
+    not have_pandas or not have_pyarrow,
+    pandas_requirement_message or pyarrow_requirement_message)
+class MaxResultArrowTests(unittest.TestCase):
+    # These tests are separate as 'spark.driver.maxResultSize' configuration
+    # is a static configuration to Spark context.
+
+    @classmethod
+    def setUpClass(cls):
+        cls.spark = SparkSession.builder \
+            .master("local[4]") \
+            .appName(cls.__name__) \
+            .config("spark.driver.maxResultSize", "10k") \
+            .getOrCreate()
+
+        # Explicitly enable Arrow and disable fallback.
+        cls.spark.conf.set("spark.sql.execution.arrow.pyspark.enabled", "true")
+        cls.spark.conf.set("spark.sql.execution.arrow.pyspark.fallback.enabled", "false")
```

Comment:
`spark.sql.execution.arrow.enabled` now has an alias `spark.sql.execution.arrow.pyspark.enabled` as of https://github.com/apache/spark/commit/d6632d185e147fcbe6724545488ad80dce20277e

And, `spark.sql.execution.arrow.pyspark.fallback.enabled` was just set to narrow down test scope. Correct behaviour should be a failure in Arrow optimized code path (although the JIRAs' case, it fails in Arrow optimized path first, and then fails again in non-Arrow optimized path).

Reply:
I think it's better to have a test that fails with default settings, for branch 2.4.

Reply:
I am going to not use aliases when I port to branch-2.4 because techincally `spark.sql.execution.arrow.enabled` and `spark.sql.execution.arrow.fallback.enabled` are supposed to be removed out in the future.

Reply:
These configurations just narrow down the testing scope, and `test_arrow.py` is supposed to test this scope (as can be seen `ArrowTests`).